packages/transforms/src/mdiIconToMdiPath/mdiIconToMdiPath.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    runTransform,
+    getParentUntilOrThrow,
+    isJsxTagElement,
+    getImportDeclarationByModuleSpecifier,
+} from '@sourcegraph/codemod-toolkit-ts'
+
+/**
+ * Convert `<Icon as={MdiIcon} />` element to `<Icon svgPath={mdiIconPath} />` component.
+ */
+export const mdiIconToMdiPath = runTransform(context => {
+    const { addManualChangeLog } = context
+
+    const mdiIconPathsToImport = new Set<string>()
+
+    const isMdiReactToken = (token: string): boolean => {
+        const importDeclaration = getImportDeclarationByModuleSpecifier(context.sourceFile, `mdi-react/${token}`)
+        return importDeclaration !== undefined
+    }
+
+    return {
+        JsxAttribute(jsxAttribute) {
+            const jsxTagElement = getParentUntilOrThrow(jsxAttribute, isJsxTagElement)
+            if (jsxTagElement.getTagNameNode().getText() !== 'Icon') {
+                // Not Icon component, so we exit
+                return
+            }
+
+            const structure = jsxAttribute.getStructure()
+            if (structure.name !== 'as' || !structure.initializer) {
+                // Not the 'as' prop or empty, so we exit
+                return
+            }
+
+            if (structure.initializer.includes(' ')) {
+                addManualChangeLog({
+                    node: jsxAttribute,
+                    message: 'Updating an expression is not supported. Please complete the transform manually',
+                })
+                // complex expression, so we exit
+                return
```

Comment:
Minor thing: we could use the `context.throwManualChangeError` helper to exit here automatically. This error type is caught in the codemod runner, so we skip only one match and continue iterating on other nodes.

P.S. I need to add some docs here 🙃

```suggestion
                throwManualChangeError({
                    node: jsxAttribute,
                    message: 'Updating an expression is not supported. Please complete the transform manually',
                })
```